src/mongo/db/pipeline/document_source_skip.cpp

```diff
@@ -84,11 +84,13 @@ Pipeline::SourceContainer::iterator DocumentSourceSkip::doOptimizeAt(
     invariant(*itr == this);
 
     auto nextSkip = dynamic_cast<DocumentSourceSkip*>((*std::next(itr)).get());
-
     if (nextSkip) {
-        _nToSkip += nextSkip->getSkip();
-        container->erase(std::next(itr));
-        return itr;
+        // '_nToSkip' can potentially overflow causing it to become negative and skip nothing
```

Comment:
All the changes look good here, thanks! Can you be sure to end all comments in periods: https://github.com/mongodb/mongo/wiki/Style-Guidelines#general-guidelines-for-commenting-code

Other than that this looks ready to go!